shared/ssl.sh

```diff
@@ -98,13 +128,20 @@ fi
 [ -s "${CSRFILE}" ] && error "certificate request '${CSRFILE}' already exists"
 openssl genrsa -out "${PRIVKEYFILE}" 4096
 openssl rsa -in "${PRIVKEYFILE}" -pubout -out "${PUBKEYFILE}"
-openssl req -new -key "${PRIVKEYFILE}" -out "${CSRFILE}" -subj "${CERTSUBJECT}"
+openssl req -new -key "${PRIVKEYFILE}" -out "${CSRFILE}" -subj "${CERTSUBJECT}" ${CERTEXTENSIONS}
 
-### Submit CSR and fail if one already exists on the CA
+### Submit CSR and fail gracefully on certain error conditions
 output="$(curl ${CURLFLAGS} -X PUT -H "Content-Type: text/plain" \
                --data-binary "@${CSRFILE}" "${CA}/certificate_request/${CERTNAME}")"
-already_exists="${CERTNAME} already has a requested certificate; ignoring certificate request"
-[ "${output}" = "${already_exists}" ] && error "unsigned CSR for '${CERTNAME}' already exists on CA"
+if [ -n "${output}" ]; then
+    cert_already_exists="${CERTNAME} already has a requested certificate; ignoring certificate request"
+    altnames_disallowed="CSR '${CERTNAME}' contains subject alternative names*"
+    case "${output}" in
+        $cert_already_exists) error "unsigned CSR for '${CERTNAME}' already exists on CA" ;;
+        $altnames_disallowed) error "DNS Alt Names not allowed by the CA" ;;
+        *) msg "[WARNING] CSR response: ${output}" ;;
+    esac
```

Comment:
😄 I'm impressed with the pattern-matching capabilities of the `sh`-compatible `case` statement 🎉